### DISCUSSION TITLE
Components with their own router.

### DISCUSSION THREAD

**Discussion by ThornWalli** [vuejs/router#1983] [Help and Questions]
Hello,

I wanted to ask if in `vue-router@4` it is still possible to give a component its own router.

In Vue@2 this could be realized by setting a new router in the skeleton of a component.

```
const router = createRouter();
export default {
  router,
  mounted() {
    console.log('Router:', this.$router);
  }
};
```

Is this still possible in Vue@3?
Background is that I have a component in a Nuxt page that gets its own router.

**Comment by posva** [comment]
This wasn't implemented in the end: https://github.com/vuejs/router/issues/1089
It's possible to reproduce locally in a variety of ways however like a custom simple set of routes with just paths and components and a custom router-view. Basically a mini routing without the complicated parts like guards, history handling, etc.
